scripts/esrupgrades/esr.5.37-7.8.mysql.up.sql

```diff
@@ -1,1295 +1,1376 @@
-				FROM
-					Channels
-					LEFT JOIN Posts FORCE INDEX (idx_posts_channel_id_update_at) ON Channels.Id = Posts.ChannelId
-				WHERE
-					Posts.RootId = ''
-				GROUP BY
-					Channels.Id) AS q ON q.channelid = Channels.Id
-				SET
-					LastRootPostAt = lastrootpost
-				WHERE
-					LastRootPostAt IS NULL;
+	-- 'ALTER TABLE PluginKeyValueStore MODIFY COLUMN PKey varchar(150);',
+	DECLARE ModifyPKey BOOLEAN;
 
-		-- sets LastRootPostAt to 0, for channels with no posts
-		UPDATE Channels SET LastRootPostAt=0 WHERE LastRootPostAt IS NULL;
+	SELECT COUNT(*) FROM Information_Schema.Columns
+		WHERE table_name = 'PluginKeyValueStore'
+		AND table_schema = DATABASE()
+		AND column_name = 'PKey'
+		AND LOWER(column_type) != 'varchar(150)'
+		INTO ModifyPKey;
+
+	IF ModifyPKey THEN
+		ALTER TABLE PluginKeyValueStore MODIFY COLUMN PKey varchar(150);
 	END IF;
 END//
 DELIMITER ;
-CALL Migrate_LastRootPostAt_Fix ();
-DROP PROCEDURE IF EXISTS Migrate_LastRootPostAt_Fix;
+SELECT CONCAT('-- ', NOW(), ' MigratePluginKeyValueStore procedure starting.') AS DEBUG;
+CALL MigratePluginKeyValueStore ();
+SELECT CONCAT('-- ', NOW(), ' MigratePluginKeyValueStore procedure finished.') AS DEBUG;
+DROP PROCEDURE IF EXISTS MigratePluginKeyValueStore;
 
-/* ==> mysql/000077_upgrade_users_v6.5.up.sql <== */
+/* ==> mysql/000078_create_oauth_mattermost_app_id.up.sql <== */
+/* ==> mysql/000082_upgrade_oauth_mattermost_app_id.up.sql <== */
+DELIMITER //
+CREATE PROCEDURE MigrateOAuthApps ()
+BEGIN
+	-- 'ALTER TABLE OAuthApps ADD COLUMN MattermostAppID varchar(32);'
+	DECLARE AddMattermostAppID BOOLEAN;
+	DECLARE AddMattermostAppIDQuery TEXT DEFAULT NULL;
 
-SET @preparedStatement = (SELECT IF(
-    (
-        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
-        WHERE table_name = 'Users'
-        AND table_schema = DATABASE()
-        AND column_name = 'AcceptedServiceTermsId'
-    ) > 0,
-    'ALTER TABLE Users DROP COLUMN AcceptedServiceTermsId;',
-    'SELECT 1'
-));
-
-PREPARE alterIfExists FROM @preparedStatement;
-EXECUTE alterIfExists;
-DEALLOCATE PREPARE alterIfExists;
+	SELECT COUNT(*) = 0 FROM INFORMATION_SCHEMA.COLUMNS
+		WHERE table_name = 'OAuthApps'
+		AND table_schema = DATABASE()
+		AND column_name = 'MattermostAppID'
+		INTO AddMattermostAppID;
 
-/* ==> mysql/000078_create_oauth_mattermost_app_id.up.sql <== */
-SET @preparedStatement = (SELECT IF(
-    (
-        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
-        WHERE table_name = 'OAuthApps'
-        AND table_schema = DATABASE()
-        AND column_name = 'MattermostAppID'
-    ) > 0,
-	'SELECT 1',
-    'ALTER TABLE OAuthApps ADD COLUMN MattermostAppID varchar(32);'
-));
-
-PREPARE alterIfExists FROM @preparedStatement;
-EXECUTE alterIfExists;
-DEALLOCATE PREPARE alterIfExists;
+	IF AddMattermostAppID THEN
+		SET AddMattermostAppIDQuery = 'ADD COLUMN MattermostAppID varchar(32) NOT NULL DEFAULT ""';
+	END IF;
+
+	IF AddMattermostAppID THEN
```

Comment:
This and the next conditional check seem like they could be folded with the one above?

Reply:
Yes, this is what happens when you take consistency to extremes :joy: Folding it.

Reply:
Done in https://github.com/mattermost/mattermost-server/pull/22590/commits/4a624c15c37b2e8a8f0b9e5577172543f37412ee.